win32/tests/threadpool_win32_int/threadpool_win32_int.c

```diff
@@ -1022,6 +1110,69 @@ static DWORD WINAPI chaos_thread_with_timers_no_lock_func(LPVOID lpThreadParamet
     return 0;
 }
 
+TEST_FUNCTION(chaos_knight_test_with_timers_no_lock_and_null_work_item)
+{
+    // start a number of threads and each of them will do a random action on the threadpool
+    EXECUTION_ENGINE_PARAMETERS execution_engine_parameters = { 4, 0 };
+    EXECUTION_ENGINE_HANDLE execution_engine = execution_engine_create(&execution_engine_parameters);
+    ASSERT_IS_NOT_NULL(execution_engine);
+    HANDLE thread_handles[CHAOS_THREAD_COUNT];
+    size_t i;
+    CHAOS_TEST_DATA chaos_test_data;
+
+    THANDLE(THREADPOOL) threadpool = threadpool_create(execution_engine);
+    ASSERT_IS_NOT_NULL(threadpool);
+    THANDLE_INITIALIZE_MOVE(THREADPOOL)(&chaos_test_data.threadpool, &threadpool);
+
+    (void)InterlockedExchange64(&chaos_test_data.expected_call_count, 0);
+    (void)InterlockedExchange64(&chaos_test_data.executed_work_functions, 0);
+    (void)InterlockedExchange64(&chaos_test_data.executed_timer_functions, 0);
+    (void)InterlockedExchange(&chaos_test_data.timers_starting, 0);
+    (void)InterlockedExchange(&chaos_test_data.chaos_test_done, 0);
+    (void)InterlockedExchange(&chaos_test_data.can_start_timers, 1);
+
+    for (i = 0; i < MAX_TIMER_COUNT; i++)
+    {
+        THANDLE_INITIALIZE(TIMER_INSTANCE)(&chaos_test_data.timers[i].timer, NULL);
+        (void)InterlockedExchange(&chaos_test_data.timers[i].state, TIMER_STATE_NONE);
+    }
+
+    chaos_test_data.work_item_context = threadpool_create_work_item(chaos_test_data.threadpool, work_function, (void*)&chaos_test_data.executed_work_functions);
+
+    for (i = 0; i < CHAOS_THREAD_COUNT; i++)
+    {
+        thread_handles[i] = CreateThread(NULL, 0, chaos_thread_with_timers_no_lock_and_null_work_item_func, &chaos_test_data, 0, NULL);
+        ASSERT_IS_NOT_NULL(thread_handles[i], "thread %zu failed to start", i);
+    }
+
+    // wait for some time
+    Sleep(TEST_RUN_TIME);
+
+    (void)InterlockedExchange(&chaos_test_data.chaos_test_done, 1);
+
+    // wait for all threads to complete
+    for (i = 0; i < CHAOS_THREAD_COUNT; i++)
+    {
+        ASSERT_IS_TRUE(WaitForSingleObject(thread_handles[i], INFINITE) == WAIT_OBJECT_0);
+    }
+
+    // assert that all scheduled items were executed
+    ASSERT_ARE_EQUAL(int64_t, (int64_t)InterlockedAdd64(&chaos_test_data.expected_call_count, 0), (int64_t)InterlockedAdd64(&chaos_test_data.executed_work_functions, 0));
+
+    LogInfo("Chaos test executed %" PRIu64 " work items, %" PRIu64 " timers",
```

Comment:
>PRIu64 [](http://example.com/codeflow?start=36&length=6)

interlocked APIs work with signed values, this should be PRId64. Please check the other places too.

Reply:
fixed